cmd/subnetcmd/upgradecmd/vm.go

```diff
@@ -0,0 +1,305 @@
+		RunE:         upgradeVM,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+	}
+
+	cmd.Flags().BoolVar(&useFuture, "future", false, "upgrade future subnet deployments")
+	cmd.Flags().BoolVar(&useLocal, "local", false, "upgrade existing `local` deployment")
+	cmd.Flags().BoolVar(&useFuji, "fuji", false, "upgrade existing `fuji` deployment (alias for `testnet`)")
+	cmd.Flags().BoolVar(&useFuji, "testnet", false, "upgrade existing `testbet` deployment (alias for `fuji`)")
+	cmd.Flags().BoolVar(&useMainnet, "mainnet", false, "upgrade existing `local` deployment")
+
+	cmd.Flags().BoolVar(&useManual, "print", false, "print instructions for upgrading")
+	cmd.Flags().StringVar(&pluginDir, "plugin-dir", "", "plugin directory to automatically upgrade VM")
+
+	cmd.Flags().BoolVar(&useLatest, "latest", false, "upgrade to latest version")
+	cmd.Flags().StringVar(&targetVersion, "version", "", "Upgrade to custom version")
+	cmd.Flags().StringVar(&useBinary, "binary", "", "Upgrade to custom binary")
+
+	return cmd
+}
+
+func atMostOneNetworkSelected() bool {
+	return !(useFuture && useLocal || useFuture && useFuji || useFuture && useMainnet || useLocal && useFuji ||
+		useLocal && useMainnet || useFuji && useMainnet)
+}
+
+func atMostOneVersionSelected() bool {
+	return !(useLatest && targetVersion != "" || useLatest && useBinary != "" || targetVersion != "" && useBinary != "")
+}
+
+func atMostOneAutomationSelected() bool {
+	return !(useManual && pluginDir != "")
+}
+
+func upgradeVM(cmd *cobra.Command, args []string) error {
+	// Check flag preconditions
+	if !atMostOneNetworkSelected() {
+		return errors.New("too many networks selected")
+	}
+
+	if !atMostOneVersionSelected() {
+		return errors.New("too many versions selected")
+	}
+
+	if !atMostOneAutomationSelected() {
+		return errors.New("--print and --plugin-dir are mutually exclusive")
+	}
+
+	subnetName := args[0]
+
+	if !app.SubnetConfigExists(subnetName) {
+		return errors.New("subnet does not exist")
+	}
+
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return fmt.Errorf("unable to load sidecar: %w", err)
+	}
+
+	networkToUpgrade, err := selectNetworkToUpgrade(sc)
+	if err != nil {
+		return err
+	}
+
+	vmType := sc.VM
+	if vmType == models.SubnetEvm || vmType == models.SpacesVM {
+		return selectUpdateOption(subnetName, vmType, sc, networkToUpgrade)
+	}
+
+	// Must be a custom update
+	return updateToCustomBin(subnetName, vmType, sc, networkToUpgrade)
+}
+
+func selectNetworkToUpgrade(sc models.Sidecar) (string, error) {
+	switch {
+	case useFuture:
+		return futureDeployment, nil
+	case useLocal:
+		return localDeployment, nil
+	case useFuji:
+		return fujiDeployment, nil
+	case useMainnet:
+		return mainnetDeployment, nil
+	}
+
+	updatePrompt := "What deployment would you like to upgrade"
+	upgradeOptions := []string{futureDeployment}
+
+	// check if subnet already deployed locally
+	locallyDeployedSubnets, err := subnet.GetLocallyDeployedSubnets(app)
```

Comment:
I think you also can check local deploy using sc.Networks ?

Reply:
I don't think so? I think we only update that for fuji and mainnet

Reply:
we don't clean this up if we run a `network clean` so I think it's more accurate to use the method I did.